pkg/volume/csi/csi_client.go

```diff
@@ -162,7 +162,7 @@ func (c *csiDriverClient) NodeGetInfo(ctx context.Context) (
 			return true, nil
 		}
 		// kubelet plugin registration service not implemented is a terminal error, no need to retry
-		if strings.Contains(getNodeInfoError.Error(), "no handler registered for plugin type") {
+		if getNodeInfoError != nil && strings.Contains(getNodeInfoError.Error(), "no handler registered for plugin type") {
```

Comment:
I'm not sure actually where "no handler registered for plugin type" comes from. I don't think it's an error from the csi client. We may be able to remove this case. cc @saad-ali 

Can we also log getNodeInfoError for the non-fatal case?

Reply:
There are 3 places in pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go where this error is generated.

Reply:
@msau42 
I have added the log for non-fatal error.